Index: src/main/java/com/opencart/qa/pagesObjects/Rubaiya.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/com/opencart/qa/pagesObjects/Rubaiya.java b/src/main/java/com/opencart/qa/pagesObjects/ReturnRequest.java
rename from src/main/java/com/opencart/qa/pagesObjects/Rubaiya.java
rename to src/main/java/com/opencart/qa/pagesObjects/ReturnRequest.java
--- a/src/main/java/com/opencart/qa/pagesObjects/Rubaiya.java	
+++ b/src/main/java/com/opencart/qa/pagesObjects/ReturnRequest.java	
@@ -1,5 +1,96 @@
 package com.opencart.qa.pagesObjects;
 
-public class Rubaiya {
+import com.opencart.qa.base.CommonAPI;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+
+public class ReturnRequest extends CommonAPI {
+
+    By MyAccount = By.xpath("//span[text()='My Account']");
+
+    By MyClickReturn = By.linkText("Returns");
+
+    By MyLogin = By.linkText("Login");
+
+    By MyEmailId = By.id("input-email");
+
+    By MyPassword = By.id("input-password");
+
+    By loginButton = By.xpath("//input[@value='Login']");
+
+    By MyRegister = By.linkText("Register");
+
+    By Continue = By.linkText("Continue");
+    By ProductReturn = By.linkText("Product Returns");
+    By ViewYourReturnRequest = By.linkText("View your return requests");
+    By MyAccountValidation = By.linkText("My Account");
+    By Desktop = By.linkText("Desktops");
+    By Mac1 = By.linkText("Mac (1)");
+    By Pc0 = By.linkText("PC (0)");
+    By PCValidation = By.linkText("PC");
+    By AddToCartMac = By.xpath("//span[text()='Add to Cart']");
+    By ValidationForAddingMac = By.className("alert alert-success alert-dismissible");
+    public WebElement ValidationForAddingMac() {
+        return driver.findElement(ValidationForAddingMac);
+    }
+
+    public WebElement AddToCartMac() {
+        return driver.findElement(AddToCartMac);
+    }
+    public WebElement PCValidation() {
+        return driver.findElement(PCValidation);
+    }
+
+    public WebElement Pc0() {
+        return driver.findElement(Pc0);
+    }
+    public WebElement Mac1() {
+        return driver.findElement(Mac1);
+    }
+    public WebElement Desktop() {
+        return driver.findElement(Desktop);
+    }
+
+    public WebElement MyAccountValidation() {
+        return driver.findElement(MyAccountValidation);
+    }
+    public WebElement ViewYourReturnRequest() {
+        return driver.findElement(ViewYourReturnRequest);
+    }
+    public WebElement ProductReturnValidation() {
+        return driver.findElement(ProductReturn);
+    }
+
+    public WebElement ContinueButton() {
+        return driver.findElement(Continue);
+    }
+    public WebElement Register() {
+        return driver.findElement(MyRegister);
+    }
+    public WebElement LoginButton() {
+        return driver.findElement(loginButton);
+    }
+
+    public WebElement Password() {
+        return driver.findElement(MyPassword);
+    }
+    public WebElement EmailId() {
+        return driver.findElement(MyEmailId);
+    }
+    public WebElement LoginField() {
+        return driver.findElement(MyLogin);
+    }
+    public WebElement AccountField() {
+        return driver.findElement(MyAccount);
+    }
+    public WebElement ClickReturn() {
+        return driver.findElement(MyClickReturn);
+    }
+
+
+
+
+
+
 
 }
Index: src/test/java/com/opencart/qa/ReturnRequestTest.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/test/java/com/opencart/qa/ReturnRequestTest.java b/src/test/java/com/opencart/qa/ReturnRequestTest.java
new file mode 100644
--- /dev/null	
+++ b/src/test/java/com/opencart/qa/ReturnRequestTest.java	
@@ -0,0 +1,82 @@
+package com.opencart.qa;
+
+import com.opencart.qa.base.CommonAPI;
+import com.opencart.qa.pagesObjects.ReturnRequest;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class ReturnRequestTest extends CommonAPI {
+
+    ReturnRequest returnRequest;
+    @Test
+    public void ReturningFromLogin() {
+
+        returnRequest = new ReturnRequest();
+        returnRequest.AccountField().click();
+        returnRequest.LoginField().click();
+        returnRequest.ClickReturn().click();
+        returnRequest.EmailId().sendKeys("onihus@gmail.com");
+        returnRequest.Password().sendKeys("onihus");
+        returnRequest.LoginButton().click();
+        Assert.assertTrue(returnRequest.ProductReturnValidation().isDisplayed());
+        returnRequest.ContinueButton().click();
+    }
+
+    @Test
+    public void ReturningFromRegister() {
+
+        returnRequest = new ReturnRequest();
+        returnRequest.AccountField().click();
+        returnRequest.Register().click();
+        returnRequest.ClickReturn().click();
+        returnRequest.EmailId().sendKeys("onihus@gmail.com");
+        returnRequest.Password().sendKeys("onihus");
+        returnRequest.LoginButton().click();
+        Assert.assertTrue(returnRequest.ProductReturnValidation().isDisplayed());
+        returnRequest.ContinueButton().click();
+    }
+
+    @Test
+    public void ViewYourReturnRequest() {
+
+        returnRequest = new ReturnRequest();
+        returnRequest.AccountField().click();
+        returnRequest.LoginField().click();
+        returnRequest.EmailId().sendKeys("onihus@gmail.com");
+        returnRequest.Password().sendKeys("onihus");
+        returnRequest.LoginButton().click();
+        returnRequest.ViewYourReturnRequest().click();
+        returnRequest.ContinueButton().click();
+        Assert.assertTrue(returnRequest.MyAccountValidation().isDisplayed());
+    }
+
+    @Test
+    public void DesktopPC() {
+
+        returnRequest = new ReturnRequest();
+        returnRequest.AccountField().click();
+        returnRequest.LoginField().click();
+        returnRequest.EmailId().sendKeys("onihus@gmail.com");
+        returnRequest.Password().sendKeys("onihus");
+        returnRequest.LoginButton().click();
+        returnRequest.Desktop().click();
+        returnRequest.Pc0().click();
+        Assert.assertTrue(returnRequest.PCValidation().isDisplayed());
+        returnRequest.ContinueButton().click();
+    }
+
+    @Test
+    public void DesktopMac() {
+
+        returnRequest = new ReturnRequest();
+        returnRequest.AccountField().click();
+        returnRequest.LoginField().click();
+        returnRequest.EmailId().sendKeys("onihus@gmail.com");
+        returnRequest.Password().sendKeys("onihus");
+        returnRequest.LoginButton().click();
+        returnRequest.Desktop().click();
+        returnRequest.Mac1().click();
+        returnRequest.AddToCartMac().click();
+        Assert.assertTrue(returnRequest.ValidationForAddingMac().isDisplayed());
+    }
+}
